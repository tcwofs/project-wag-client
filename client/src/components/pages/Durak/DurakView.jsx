import { Button, Divider, Grid, Paper, Typography } from '@material-ui/core';
import React, { useEffect, useState } from 'react';
import io from 'socket.io-client';
import { useStyles } from './DurakView.styles';

let socket;

export default props => {
  const classes = useStyles();
  const [finished, setFinished] = useState(null);
  const [lobby, setLobby] = useState(true);
  const [users, setUsers] = useState([]);
  const [userhand, setUserhand] = useState([]);
  const [lastcard, setLastcard] = useState();
  const [otherusers, setOtherusers] = useState();
  const [cardcount, setCardcount] = useState();
  const [userstatus, setUserStatus] = useState('other');
  const [field, setField] = useState();

  if (!props.location.state) {
    window.location.href = `http://${window.location.host}/`;
  }
  const { username, roomname } = props.location.state;
  const ENDPOINT = 'http://' + window.location.host + '/durak';

  useEffect(() => {
    socket = io(ENDPOINT);
    socket.emit('connect-room', { roomname });

    return () => {
      socket.emit('disconnect');
      socket.off();
    };
  }, [ENDPOINT, username, roomname]);

  useEffect(() => {
    socket.on('error-redirect', message => {
      window.alert(message.error);
      window.location.href = `http://${window.location.host}/`;
    });
    socket.on('error-msg', message => {
      window.alert(message.error);
    });
  }, []);

  useEffect(() => {
    socket.on('get-room-users', ({ activeUsers }) => {
      setUsers(activeUsers);
    });

    socket.on('start-game', () => {
      setLobby(false);
    });

    socket.on('handcards', ({ recievedUserhand, allcards, userhands, status }) => {
      setUserhand(recievedUserhand);
      setLastcard(allcards.lastcard);
      setCardcount(allcards.cardcount);
      setField(allcards.field);
      setOtherusers(userhands);
      setUserStatus(status);
    });

    socket.on('update-field', ({ updatedField, updatedUserhand }) => {
      setField(updatedField);
      setUserhand(updatedUserhand);
    });

    socket.on('finish-game', ({ lostuser }) => {
      setFinished(lostuser.username);
    });
  });

  const userReady = () => {
    socket.emit('user-ready', { roomname });
  };

  const cardAttack = ({ card, second }) => {
    socket.emit('attack', { card, roomname, second });
  };

  const cardDeffence = ({ card }) => {
    socket.emit('defence', { card, roomname });
  };

  const finishMove = () => {
    socket.emit('finish-attack');
  };

  return (
    <div className={classes.main}>
      <Paper className={classes.paper}>
        <Typography style={{ textAlign: 'center' }} variant='h4'>
          {roomname}
        </Typography>
        <Divider style={{ marginBottom: '1rem' }} light />
        {finished !== null ? (
          <div>
            <p style={{ textAlign: 'center' }}>{finished} lost! game ended</p>
            <Button variant='contained' color='secondary' href={`http://${window.location.host}/`}>
              Go Home
            </Button>
          </div>
        ) : (
          [
            lobby ? (
              <div key={1}>
                {users.map(user => (
                  <p key={user.id}>
                    {user.username} | {user.ready ? 'ready' : 'not ready'}
                  </p>
                ))}
                <Button variant='contained' color='secondary' onClick={userReady}>
                  <Typography>Ready</Typography>
                </Button>
              </div>
            ) : (
              <div className={classes.gamefield} key={1}>
                <Grid container spacing={2} alignItems='flex-start' justify='flex-end' direction='row'>
                  {otherusers ? (
                    otherusers.map(othuser => (
                      <Grid item xs={12 / otherusers.length}>
                        <img
                          alt=''
                          style={{
                            maxWidth: '4rem',
                            width: '80%',
                            height: 'auto',
                          }}
                          src={process.env.PUBLIC_URL + `/cards/gray_back.png`}
                        />
                        <div>
                          <Typography>{othuser.username}</Typography>
                          <Typography>{othuser.handlength}</Typography>
                        </div>
                      </Grid>
                    ))
                  ) : (
                    <></>
                  )}
                </Grid>

                <Grid container spacing={2}>
                  <Grid item xs={9}>
                    <Grid container spacing={1}>
                      {field &&
                        field.map(row => (
                          <Grid item xs={2}>
                            {row.map(item => (
                              <img
                                alt=''
                                style={{
                                  maxWidth: '6rem',
                                  width: '100%',
                                  height: 'auto',
                                }}
                                src={process.env.PUBLIC_URL + `/cards/${item}.png`}
                              />
                            ))}
                          </Grid>
                        ))}
                    </Grid>
                  </Grid>
                  <Grid item xs={3} style={{ position: 'relative' }}>
                    <img
                      alt=''
                      style={{
                        position: 'relative',
                        maxWidth: '6rem',
                        width: '100%',
                        height: 'auto',
                      }}
                      src={process.env.PUBLIC_URL + `/cards/${lastcard}.png`}
                    />
                    <img
                      alt=''
                      style={{
                        transform: 'rotate(90deg)',
                        maxWidth: '6rem',
                        width: '80%',
                        height: 'auto',
                        position: 'absolute',
                        top: '30px',
                        left: '20px',
                      }}
                      src={process.env.PUBLIC_URL + `/cards/gray_back.png`}
                    />
                    <div>
                      <Typography>{cardcount}</Typography>
                      {userstatus === 'other' || !userstatus ? (
                        <Typography>Please wait for you turn</Typography>
                      ) : userstatus === 'attacking_1' ? (
                        <div>
                          <Typography>You are attacking 1st</Typography>
                          <Button variant='contained' color='secondary' onClick={finishMove}>
                            <Typography>Finish</Typography>
                          </Button>
                        </div>
                      ) : userstatus === 'attacking_2' ? (
                        <div>
                          <Typography>You are attacking 2nd</Typography>
                          <Button variant='contained' color='secondary' onClick={finishMove}>
                            <Typography>Finish</Typography>
                          </Button>
                        </div>
                      ) : (
                        <div>
                          <Typography>You are defending</Typography>
                          <Button variant='contained' color='secondary' onClick={finishMove}>
                            {field.filter(row => row.length % 2 === 0).length === field.length ? (
                              <Typography>Draft</Typography>
                            ) : (
                              <Typography>Take</Typography>
                            )}
                          </Button>
                        </div>
                      )}
                    </div>
                  </Grid>
                  <Grid item xs={12}>
                    <Grid container spacing={0}>
                      {userhand &&
                        userhand.map(card => (
                          <Grid item xs={2}>
                            <div>
                              <img
                                alt=''
                                style={{ maxWidth: '6rem', width: '100%', height: 'auto' }}
                                src={process.env.PUBLIC_URL + `/cards/${card}.png`}
                                onClick={e => {
                                  userstatus === 'attacking_1'
                                    ? cardAttack({ card, second: false })
                                    : userstatus === 'attacking_2'
                                    ? cardAttack({ card, second: true })
                                    : userstatus === 'defending'
                                    ? cardDeffence({ card })
                                    : e.preventDefault();
                                }}
                              />
                            </div>
                          </Grid>
                        ))}
                    </Grid>
                  </Grid>
                </Grid>
              </div>
            ),
          ]
        )}
      </Paper>
    </div>
  );
};
