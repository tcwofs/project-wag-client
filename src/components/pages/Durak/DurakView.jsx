import { Button, Grid, Paper, Typography } from '@material-ui/core';
import React, { useContext, useEffect, useState } from 'react';
import { Link as RouterLink, useParams } from 'react-router-dom';
import { v4 as uuidv4 } from 'uuid';
import { SocketContext } from '../../../app';
import { useStyles } from './DurakView.styles';

const DurakView = () => {
  const { socketMain } = useContext(SocketContext);
  const classes = useStyles();
  const [finishgame, setFinishgame] = useState(null);
  const [userhand, setUserhand] = useState([]);
  const [lastcard, setLastcard] = useState();
  const [otherusers, setOtherusers] = useState();
  const [cardcount, setCardcount] = useState();
  const [userstatus, setUserStatus] = useState('other');
  const [field, setField] = useState();
  const [userfinished, setUserfinished] = useState(false);
  const { roomId } = useParams();

  useEffect(() => {
    socketMain.on('error-redirect', (message) => {
      window.alert(message.error);
      window.location.href = `http://${window.location.host}/`;
    });
    socketMain.on('error-msg', (message) => {
      window.alert(message.error);
    });
    return () => {
      socketMain.off();
    };
  }, [socketMain]);

  useEffect(() => {
    socketMain.on('handcards', ({ recievedUserhand, allcards, userhands, status, finished }) => {
      setUserhand(recievedUserhand);
      setLastcard(allcards.lastcard);
      setCardcount(allcards.cardcount);
      setField(allcards.field);
      setOtherusers(userhands);
      setUserStatus(status);
      setUserfinished(finished);
    });

    socketMain.on('update-field', ({ updatedField, updatedUserhand }) => {
      setField(updatedField);
      setUserhand(updatedUserhand);
    });

    socketMain.on('finish-game', ({ lostuser }) => {
      setFinishgame(lostuser.username);
    });

    socketMain.on('play-again', () => {
      setFinishgame(null);
    });
  });

  const cardAttack = ({ card, second }) => {
    socketMain.emit('attack', { card, roomId, second });
  };

  const cardDeffence = ({ card }) => {
    socketMain.emit('defence', { card, roomId });
  };

  const finishMove = () => {
    setUserfinished(!userfinished);
    socketMain.emit('finish-move', { roomId });
  };

  const playAgain = () => {
    socketMain.emit('play-again', { roomId });
  };

  const goHome = () => {
    socketMain.emit('durak-leave');
  };

  return (
    <div className={classes.main}>
      <Paper className={classes.paper}>
        <Button component={RouterLink} to={`/`} size='small' variant='outlined' color='secondary' style={{ float: 'right' }} onClick={goHome}>
          home
        </Button>
        {finishgame !== null ? (
          <div>
            <p style={{ textAlign: 'center' }}>{finishgame} lost! game ended</p>
            <Button variant='contained' color='secondary' onClick={playAgain}>
              Play Again
            </Button>
          </div>
        ) : (
          <div className={classes.gamefield} key={1}>
            <Grid container spacing={2} alignItems='flex-start' justify='flex-end' direction='row'>
              {otherusers ? (
                otherusers.map((othuser) => (
                  <Grid item xs={12 / otherusers.length} key={uuidv4()}>
                    <img
                      alt=''
                      style={{
                        maxWidth: '4rem',
                        width: '80%',
                        height: 'auto',
                      }}
                      src={process.env.PUBLIC_URL + '/cards/gray_back.png'}
                    />
                    <div>
                      <Typography>{othuser.username}</Typography>
                      <Typography>{othuser.handlength}</Typography>
                    </div>
                  </Grid>
                ))
              ) : (
                <></>
              )}
            </Grid>

            <Grid container spacing={2}>
              <Grid item xs={9}>
                <Grid container spacing={1}>
                  {field &&
                    field.map((row) => (
                      <Grid item xs={2} key={uuidv4()}>
                        {row.map((item) => (
                          <img
                            key={uuidv4()}
                            alt=''
                            style={{
                              maxWidth: '6rem',
                              width: '100%',
                              height: 'auto',
                            }}
                            src={process.env.PUBLIC_URL + `/cards/${item}.png`}
                          />
                        ))}
                      </Grid>
                    ))}
                </Grid>
              </Grid>
              <Grid item xs={3} style={{ position: 'relative' }}>
                <img
                  alt=''
                  style={{
                    position: 'relative',
                    maxWidth: '6rem',
                    width: '100%',
                    height: 'auto',
                  }}
                  src={process.env.PUBLIC_URL + `/cards/${lastcard}.png`}
                />
                <img
                  alt=''
                  style={{
                    transform: 'rotate(90deg)',
                    maxWidth: '6rem',
                    width: '80%',
                    height: 'auto',
                    position: 'absolute',
                    top: '30px',
                    left: '20px',
                  }}
                  src={process.env.PUBLIC_URL + '/cards/gray_back.png'}
                />
                <div>
                  <Typography>{cardcount}</Typography>
                  {userstatus === 'other' || !userstatus ? (
                    <Typography>Please wait for you turn</Typography>
                  ) : userstatus === 'attacking_1' ? (
                    <div>
                      <Typography>You are attacking 1st</Typography>
                      <Button variant='contained' color='secondary' onClick={finishMove}>
                        {!userfinished ? <Typography>Finish</Typography> : <Typography>Cancel</Typography>}
                      </Button>
                    </div>
                  ) : userstatus === 'attacking_2' ? (
                    <div>
                      <Typography>You are attacking 2nd</Typography>
                      <Button variant='contained' color='secondary' onClick={finishMove}>
                        {!userfinished ? <Typography>Finish</Typography> : <Typography>Cancel</Typography>}
                      </Button>
                    </div>
                  ) : (
                    <div>
                      <Typography>You are defending</Typography>
                      <Button variant='contained' color='secondary' onClick={finishMove}>
                        {field.filter((row) => row.length % 2 === 0).length === field.length ? (
                          !userfinished ? (
                            <Typography>Draft</Typography>
                          ) : (
                            <Typography>Cancel</Typography>
                          )
                        ) : !userfinished ? (
                          <Typography>Take</Typography>
                        ) : (
                          <Typography>Cancel</Typography>
                        )}
                      </Button>
                    </div>
                  )}
                </div>
              </Grid>
              <Grid item xs={12}>
                <Grid container spacing={0}>
                  {userhand &&
                    userhand.map((card) => (
                      <Grid item xs={2} md={1} key={uuidv4()}>
                        <div>
                          <img
                            alt=''
                            style={{ maxWidth: '6rem', width: '100%', height: 'auto' }}
                            src={process.env.PUBLIC_URL + `/cards/${card}.png`}
                            onClick={(e) => {
                              userstatus === 'attacking_1'
                                ? cardAttack({ card, second: false })
                                : userstatus === 'attacking_2'
                                ? cardAttack({ card, second: true })
                                : userstatus === 'defending'
                                ? cardDeffence({ card })
                                : e.preventDefault();
                            }}
                          />
                        </div>
                      </Grid>
                    ))}
                </Grid>
              </Grid>
            </Grid>
          </div>
        )}
      </Paper>
    </div>
  );
};

export default DurakView;
