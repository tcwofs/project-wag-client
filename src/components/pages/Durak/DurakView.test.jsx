import { createMount } from '@material-ui/core/test-utils';
import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import DurakView from './DurakView';

let mount;
let wrapper;
let location;

describe('<DurakViFew />', () => {
  beforeEach(() => {
    mount = createMount();
    location = {
      state: {
        username: 'Test',
        roomname: 'Room',
      },
    };
  });

  afterEach(() => {
    mount.cleanUp();
  });

  it('should work', () => {
    wrapper = mount(<DurakView location={location} />);
  });

  test('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
