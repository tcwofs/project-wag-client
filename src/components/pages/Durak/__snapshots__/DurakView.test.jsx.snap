// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DurakViFew /> renders correctly 1`] = `
<DurakView
  location={
    Object {
      "state": Object {
        "roomname": "Room",
        "username": "Test",
      },
    }
  }
>
  <div
    className="makeStyles-main-1"
  >
    <WithStyles(ForwardRef(Paper))
      className="makeStyles-paper-2"
    >
      <ForwardRef(Paper)
        className="makeStyles-paper-2"
        classes={
          Object {
            "elevation0": "MuiPaper-elevation0",
            "elevation1": "MuiPaper-elevation1",
            "elevation10": "MuiPaper-elevation10",
            "elevation11": "MuiPaper-elevation11",
            "elevation12": "MuiPaper-elevation12",
            "elevation13": "MuiPaper-elevation13",
            "elevation14": "MuiPaper-elevation14",
            "elevation15": "MuiPaper-elevation15",
            "elevation16": "MuiPaper-elevation16",
            "elevation17": "MuiPaper-elevation17",
            "elevation18": "MuiPaper-elevation18",
            "elevation19": "MuiPaper-elevation19",
            "elevation2": "MuiPaper-elevation2",
            "elevation20": "MuiPaper-elevation20",
            "elevation21": "MuiPaper-elevation21",
            "elevation22": "MuiPaper-elevation22",
            "elevation23": "MuiPaper-elevation23",
            "elevation24": "MuiPaper-elevation24",
            "elevation3": "MuiPaper-elevation3",
            "elevation4": "MuiPaper-elevation4",
            "elevation5": "MuiPaper-elevation5",
            "elevation6": "MuiPaper-elevation6",
            "elevation7": "MuiPaper-elevation7",
            "elevation8": "MuiPaper-elevation8",
            "elevation9": "MuiPaper-elevation9",
            "outlined": "MuiPaper-outlined",
            "root": "MuiPaper-root",
            "rounded": "MuiPaper-rounded",
          }
        }
      >
        <div
          className="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation1 MuiPaper-rounded"
        >
          <WithStyles(ForwardRef(Typography))
            style={
              Object {
                "textAlign": "center",
              }
            }
            variant="h4"
          >
            <ForwardRef(Typography)
              classes={
                Object {
                  "alignCenter": "MuiTypography-alignCenter",
                  "alignJustify": "MuiTypography-alignJustify",
                  "alignLeft": "MuiTypography-alignLeft",
                  "alignRight": "MuiTypography-alignRight",
                  "body1": "MuiTypography-body1",
                  "body2": "MuiTypography-body2",
                  "button": "MuiTypography-button",
                  "caption": "MuiTypography-caption",
                  "colorError": "MuiTypography-colorError",
                  "colorInherit": "MuiTypography-colorInherit",
                  "colorPrimary": "MuiTypography-colorPrimary",
                  "colorSecondary": "MuiTypography-colorSecondary",
                  "colorTextPrimary": "MuiTypography-colorTextPrimary",
                  "colorTextSecondary": "MuiTypography-colorTextSecondary",
                  "displayBlock": "MuiTypography-displayBlock",
                  "displayInline": "MuiTypography-displayInline",
                  "gutterBottom": "MuiTypography-gutterBottom",
                  "h1": "MuiTypography-h1",
                  "h2": "MuiTypography-h2",
                  "h3": "MuiTypography-h3",
                  "h4": "MuiTypography-h4",
                  "h5": "MuiTypography-h5",
                  "h6": "MuiTypography-h6",
                  "noWrap": "MuiTypography-noWrap",
                  "overline": "MuiTypography-overline",
                  "paragraph": "MuiTypography-paragraph",
                  "root": "MuiTypography-root",
                  "srOnly": "MuiTypography-srOnly",
                  "subtitle1": "MuiTypography-subtitle1",
                  "subtitle2": "MuiTypography-subtitle2",
                }
              }
              style={
                Object {
                  "textAlign": "center",
                }
              }
              variant="h4"
            >
              <h4
                className="MuiTypography-root MuiTypography-h4"
                style={
                  Object {
                    "textAlign": "center",
                  }
                }
              >
                Room
              </h4>
            </ForwardRef(Typography)>
          </WithStyles(ForwardRef(Typography))>
          <WithStyles(ForwardRef(Divider))
            light={true}
            style={
              Object {
                "marginBottom": "1rem",
              }
            }
          >
            <ForwardRef(Divider)
              classes={
                Object {
                  "absolute": "MuiDivider-absolute",
                  "flexItem": "MuiDivider-flexItem",
                  "inset": "MuiDivider-inset",
                  "light": "MuiDivider-light",
                  "middle": "MuiDivider-middle",
                  "root": "MuiDivider-root",
                  "vertical": "MuiDivider-vertical",
                }
              }
              light={true}
              style={
                Object {
                  "marginBottom": "1rem",
                }
              }
            >
              <hr
                className="MuiDivider-root MuiDivider-light"
                style={
                  Object {
                    "marginBottom": "1rem",
                  }
                }
              />
            </ForwardRef(Divider)>
          </WithStyles(ForwardRef(Divider))>
          <div
            key="1"
          >
            <WithStyles(ForwardRef(Button))
              color="secondary"
              onClick={[Function]}
              variant="contained"
            >
              <ForwardRef(Button)
                classes={
                  Object {
                    "colorInherit": "MuiButton-colorInherit",
                    "contained": "MuiButton-contained",
                    "containedPrimary": "MuiButton-containedPrimary",
                    "containedSecondary": "MuiButton-containedSecondary",
                    "containedSizeLarge": "MuiButton-containedSizeLarge",
                    "containedSizeSmall": "MuiButton-containedSizeSmall",
                    "disableElevation": "MuiButton-disableElevation",
                    "disabled": "Mui-disabled",
                    "endIcon": "MuiButton-endIcon",
                    "focusVisible": "Mui-focusVisible",
                    "fullWidth": "MuiButton-fullWidth",
                    "iconSizeLarge": "MuiButton-iconSizeLarge",
                    "iconSizeMedium": "MuiButton-iconSizeMedium",
                    "iconSizeSmall": "MuiButton-iconSizeSmall",
                    "label": "MuiButton-label",
                    "outlined": "MuiButton-outlined",
                    "outlinedPrimary": "MuiButton-outlinedPrimary",
                    "outlinedSecondary": "MuiButton-outlinedSecondary",
                    "outlinedSizeLarge": "MuiButton-outlinedSizeLarge",
                    "outlinedSizeSmall": "MuiButton-outlinedSizeSmall",
                    "root": "MuiButton-root",
                    "sizeLarge": "MuiButton-sizeLarge",
                    "sizeSmall": "MuiButton-sizeSmall",
                    "startIcon": "MuiButton-startIcon",
                    "text": "MuiButton-text",
                    "textPrimary": "MuiButton-textPrimary",
                    "textSecondary": "MuiButton-textSecondary",
                    "textSizeLarge": "MuiButton-textSizeLarge",
                    "textSizeSmall": "MuiButton-textSizeSmall",
                  }
                }
                color="secondary"
                onClick={[Function]}
                variant="contained"
              >
                <WithStyles(ForwardRef(ButtonBase))
                  className="MuiButton-root MuiButton-contained MuiButton-containedSecondary"
                  component="button"
                  disabled={false}
                  focusRipple={true}
                  focusVisibleClassName="Mui-focusVisible"
                  onClick={[Function]}
                  type="button"
                >
                  <ForwardRef(ButtonBase)
                    className="MuiButton-root MuiButton-contained MuiButton-containedSecondary"
                    classes={
                      Object {
                        "disabled": "Mui-disabled",
                        "focusVisible": "Mui-focusVisible",
                        "root": "MuiButtonBase-root",
                      }
                    }
                    component="button"
                    disabled={false}
                    focusRipple={true}
                    focusVisibleClassName="Mui-focusVisible"
                    onClick={[Function]}
                    type="button"
                  >
                    <button
                      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
                      disabled={false}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={0}
                      type="button"
                    >
                      <span
                        className="MuiButton-label"
                      >
                        <WithStyles(ForwardRef(Typography))>
                          <ForwardRef(Typography)
                            classes={
                              Object {
                                "alignCenter": "MuiTypography-alignCenter",
                                "alignJustify": "MuiTypography-alignJustify",
                                "alignLeft": "MuiTypography-alignLeft",
                                "alignRight": "MuiTypography-alignRight",
                                "body1": "MuiTypography-body1",
                                "body2": "MuiTypography-body2",
                                "button": "MuiTypography-button",
                                "caption": "MuiTypography-caption",
                                "colorError": "MuiTypography-colorError",
                                "colorInherit": "MuiTypography-colorInherit",
                                "colorPrimary": "MuiTypography-colorPrimary",
                                "colorSecondary": "MuiTypography-colorSecondary",
                                "colorTextPrimary": "MuiTypography-colorTextPrimary",
                                "colorTextSecondary": "MuiTypography-colorTextSecondary",
                                "displayBlock": "MuiTypography-displayBlock",
                                "displayInline": "MuiTypography-displayInline",
                                "gutterBottom": "MuiTypography-gutterBottom",
                                "h1": "MuiTypography-h1",
                                "h2": "MuiTypography-h2",
                                "h3": "MuiTypography-h3",
                                "h4": "MuiTypography-h4",
                                "h5": "MuiTypography-h5",
                                "h6": "MuiTypography-h6",
                                "noWrap": "MuiTypography-noWrap",
                                "overline": "MuiTypography-overline",
                                "paragraph": "MuiTypography-paragraph",
                                "root": "MuiTypography-root",
                                "srOnly": "MuiTypography-srOnly",
                                "subtitle1": "MuiTypography-subtitle1",
                                "subtitle2": "MuiTypography-subtitle2",
                              }
                            }
                          >
                            <p
                              className="MuiTypography-root MuiTypography-body1"
                            >
                              Ready
                            </p>
                          </ForwardRef(Typography)>
                        </WithStyles(ForwardRef(Typography))>
                      </span>
                      <NoSsr>
                        <WithStyles(memo)
                          center={false}
                        >
                          <ForwardRef(TouchRipple)
                            center={false}
                            classes={
                              Object {
                                "child": "MuiTouchRipple-child",
                                "childLeaving": "MuiTouchRipple-childLeaving",
                                "childPulsate": "MuiTouchRipple-childPulsate",
                                "ripple": "MuiTouchRipple-ripple",
                                "ripplePulsate": "MuiTouchRipple-ripplePulsate",
                                "rippleVisible": "MuiTouchRipple-rippleVisible",
                                "root": "MuiTouchRipple-root",
                              }
                            }
                          >
                            <span
                              className="MuiTouchRipple-root"
                            >
                              <TransitionGroup
                                childFactory={[Function]}
                                component={null}
                                exit={true}
                              />
                            </span>
                          </ForwardRef(TouchRipple)>
                        </WithStyles(memo)>
                      </NoSsr>
                    </button>
                  </ForwardRef(ButtonBase)>
                </WithStyles(ForwardRef(ButtonBase))>
              </ForwardRef(Button)>
            </WithStyles(ForwardRef(Button))>
          </div>
        </div>
      </ForwardRef(Paper)>
    </WithStyles(ForwardRef(Paper))>
  </div>
</DurakView>
`;
